### PR TITLE
Native avatars and branding bits

### PR DESCRIPTION
**Pull request by juliushaertl**
Companion pr for https://github.com/CollaboraOnline/online/pull/3677

- Improve loading behavior in viewer
- Handle richdocumentscode checking
- use native avatar list
- Introduce 
- Branding bits

### CODE REVIEW

File: src/view/Office.vue [hunk 1]
```diff
@@ -22,9 +22,27 @@
 
 <template>
 	<transition name="fade" appear>
-		<div v-show="!loading" id="richdocuments-wrapper">
-			<div class="header">
-				<!-- This is obviously not the way to go since it would require absolute positioning and therefore not be compatible with viewer actions/sidebar -->
+		<div id="richdocuments-wrapper">
+			<div v-if="showLoadingIndicator" id="cool-loading-overlay" :class="{ debug: debug }">
```
Comment (eneiluj) [inline src/view/Office.vue:26]: Small detail but: Isn't it more readable when each attribute is on a new line?

File: src/view/Office.vue [hunk 2]
```diff
@@ -111,7 +178,37 @@ export default {
...
+		documentReady() {
+			this.loading = LOADING_STATE.DOCUMENT_READY
+			clearTimeout(this.loadingTimeout)
+		},
+		async share() {
+			FilesAppIntegration.share()
+		},
+		close() {
+			this.$parent.close()
+		},
```
Comment (eneiluj) [inline src/view/Office.vue:210]: Couldn't this be done with an event to make it more robust?

**Review by eneiluj — COMMENTED** [review]
I didn't grasp everything but the code looks good.

The avatar list is beautiful.

After having closed the Files sidebar while editing a document, I can't open it again by pressing the icon-menu-sidebar. I get this in the browser logs:

```
document.js:348 [document] Received post message  {msgId: 'UI_Share', args: {…}, deprecated: false}
postMessage.tsx:43 PostMessageService.sendPostMessage parent {"MessageId":"UI_Share","SendTime":1638210804578,"Values":{}}
document.js:470 [document] Unhandled post message {msgId: 'UI_Share', args: {…}, deprecated: false}
Office.vue?vue&type=script&lang=js&:238 [viewer] Received post message {msgId: 'UI_Share', args: {…}, deprecated: false}
```
I'm using the latest collabora/code docker image.

File: src/view/Office.vue (hunk 1 above)
Comment (juliushaertl) [reply src/view/Office.vue:26]: Up to 3 are allowed for a single line, but agreed that it is more readable. I'll split that.

File: src/view/Office.vue (hunk 2 above)
Comment (juliushaertl) [reply src/view/Office.vue:210]: Would need support in the viewer first, but might be a good idea to introduce that there.

**Comment by juliushaertl** [comment]
Thanks for catching that, all addressed.

**Review by luka-nextcloud — APPROVED** [review]

**Review by eneiluj — APPROVED** [review]
:+1:

**Pull request merged**